### PR TITLE
Reset length on data type change

### LINKED ISSUE #2528: Changing IM data type problem
Current:

- set `VARCHAR(15)` -> save -> set `DATE` -> save -> you'll see `DATE(15)`

Expected

- set `VARCHAR(15)` -> save -> set `DATE` -> save -> you'll see `DATE`

Length attr should be deleted

### PR DESCRIPTION
**Pull request by ujovlado**
Fixes #2528

Proposed changes:

- length will be set to null on type change

### CODE REVIEW

**Comment by ujovlado** [comment]
https://keboola.zendesk.com/agent/tickets/9548

File: src/scripts/modules/transformations/react/components/mapping/input/DatatypeFormRow.jsx [hunk 1]
```diff
@@ -27,7 +27,11 @@ export default React.createClass({
 
   handleTypeChange(newType) {
     if (newType) {
-      this.props.onChange(this.props.datatype.set('type', newType.value));
+      this.props.onChange(
+        this.props.datatype
+          .set('length', null)
```
Comment (kacurez) [inline src/scripts/modules/transformations/react/components/mapping/input/DatatypeFormRow.jsx]: ciste zo zvedavosti - to nerozbije nic ked sa posle `length: null` na "backend"?

Comment (kacurez) [reply src/scripts/modules/transformations/react/components/mapping/input/DatatypeFormRow.jsx]: ci to sa pri `date` asi ignoruje co?

Comment (ujovlado) [reply src/scripts/modules/transformations/react/components/mapping/input/DatatypeFormRow.jsx]: pozera som a napr. `INTEGER` datovy typ ma `length` `null`, takze imho nie

**Pull request merged**